Drop previous_response_id in favor of Conversations API

raghubetina opened the pull request:
OpenAI's Conversations API doesn't support forking - `conversation_id` and `previous_response_id` are mutually exclusive parameters. Since the gem always creates conversations, `previous_response_id` was never being used effectively.

This simplifies the gem to focus solely on the Conversations API pattern where conversation_id maintains context across turns.

Resolves #38 

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Remove `previous_response_id` in favor of `conversation_id` for maintaining context in AI::Chat.
> 
>   - **Behavior**:
>     - Removes `previous_response_id` functionality in `lib/ai/chat.rb` and examples, focusing on `conversation_id` for context.
>     - Updates `generate!` and `get_response` methods to use `@last_response_id` instead of `previous_response_id`.
>   - **Documentation**:
>     - Removes sections related to `previous_response_id` from `README.md`.
>     - Updates examples in `examples/01_quick.rb`, `examples/02_core.rb`, and `examples/09_edge_cases.rb` to reflect the removal of `previous_response_id`.
>   - **Tests**:
>     - Removes integration tests related to `previous_response_id` in `spec/integration/ai_chat_integration_spec.rb`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=firstdraft%2Fai-chat&utm_source=github&utm_medium=referral)<sup> for d68de21a9665a35f2af6d5b20562ee61d1f63b3d. You can [customize](https://app.ellipsis.dev/firstdraft/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

raghubetina committed:
Drop previous_response_id for Conversations API

OpenAI's Conversations API doesn't support forking - conversation_id
and previous_response_id are mutually exclusive parameters. Since the
gem always creates conversations, previous_response_id was never being
used effectively.

This simplifies the gem to focus solely on the Conversations API
pattern where conversation_id maintains context across turns.

raghubetina commented:
@codex Review

jelaniwoods reviewed (approved):
I did notice that `previous_response_id` was only really being used for background mode while working on proxy integration. I had similar thoughts about its usefulness.

The changes LGTM.

bpurinton commented on examples/13_conversation_features_comprehensive.rb:
```diff
@@ -5,8 +5,6 @@
 # - Conversation continuity across multiple turns
 # - Inspecting conversation items (programmatically and verbose)
 # - Loading existing conversations
```
while running the examples, I noticed a typo for this file in `examples/all.rb`. Currently, it is written in that file as:

```rb
# ...
# Conversations
require_relative "15_conversations_features_comprehensive"
# ...
```

It should be:

```rb
# ...
# Conversations
require_relative "15_conversation_features_comprehensive"
# ...
```

(dropping the `s` from that file, or changing the filename here, whichever you prefer)

jelaniwoods replied:
@bpurinton this is fixed in https://github.com/firstdraft/ai-chat/pull/41

bpurinton replied:
Again unrelated to this PR (maybe?), but worth fixing, `examples/16_schema_generation.rb` throws an error on the last example:

```rb
# ...

puts "7. AI::Chat can load the schema from a given file"
puts "-" * 60
puts
puts AI::Chat.generate_schema!("A user with full name (required), first_name (required), and last_name (required) default 'Smith'.")
puts
chat = AI::Chat.new
chat.schema_file = "schema.json"
if chat.schema.present?
  puts "✓ AI::Chat#schema_file= assigns AI::Chat#schema"
else
  puts "✗ AI::Chat#schema_file= does not assign AI::Chat#schema"
end
puts
```

The problem line is `chat.schema.present?`, which throws:

```rb
examples/16_schema_generation.rb:112:in `<main>': undefined method `present?' for {:format=>{:name=>"user_with_default_last_name", :strict=>true, :schema=>{:type=>"object", :description=>"User object with full name, first name, and last name (defaulting to 'Smith').", :properties=>{:full_name=>{:type=>"string", :description=>"The user's full name."}, :first_name=>{:type=>"string", :description=>"The user's given name."}, :last_name=>{:type=>"string", :description=>"The user's family name. Defaults to 'Smith' if not otherwise specified.", :default=>"Smith"}}, :required=>["full_name", "first_name", "last_name"], :additionalProperties=>false}, :type=>:json_schema}}:Hash (NoMethodError)

if chat.schema.present?
              ^^^^^^^^^
```

Instead, you can change that line to:

```rb
if chat.schema
```

And use the truthiness of the that column to determine the result.

bpurinton reviewed (approved):
These changes LGTM after running examples. Minor issues I noted with examples are fixed on other PRs (but worth a double check)

raghubetina committed:
Cleanup and simplification (#46)

Default configuration:
- Model: gpt-5.1 (was gpt-4.1-nano)
- reasoning_effort: nil (equivalent to "none" for gpt-5.1)
- All tools (web_search, code_interpreter, image_generation) now
  work out of the box without configuration

Code changes:
- Add last_response_id as public reader for background mode workflows
- Change web_search_preview tool to web_search (matches OpenAI GA)
- Use last_response_id accessor internally instead of @instance_var

Test improvements:
- Add tests for conversation continuity across generate! calls
- Add tests for continuing conversations in new instances
- Add tests for file handling and conversation items retrieval
- Add test for last_response_id updates
- Remove low-value and redundant tests

Documentation:
- Update CHANGELOG.md with 0.4.0 changes
- Update DEVELOPMENT_JOURNEY.md helper methods section
- Simplify README by removing tool compatibility warnings
- Add note that nil reasoning equals "none" for gpt-5.1

Examples:
- Renumber example files (now 01-15)
- Remove redundant/low-value examples (03_configuration,
  08_advanced_usage, tmp_image)
- Remove reasoning_effort=nil lines (no longer needed)

Infrastructure:
- Fix Gemfile.lock to use openai gem from RubyGems
- Delete local openai-ruby reference directory
